x/superfluid/keeper/stake.go

```diff
@@ -323,59 +315,64 @@ func (k Keeper) SuperfluidDelegate(ctx sdk.Context, sender string, lockID uint64
 	return nil
 }
 
-func (k Keeper) SuperfluidUndelegate(ctx sdk.Context, sender string, lockID uint64) (sdk.ValAddress, error) {
+func (k Keeper) SuperfluidUndelegate(ctx sdk.Context, sender string, lockID uint64) error {
 	lock, err := k.lk.GetLockByID(ctx, lockID)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
 	if lock.Owner != sender {
-		return nil, lockuptypes.ErrNotLockOwner
+		return lockuptypes.ErrNotLockOwner
 	}
 
 	// Remove previously created synthetic lockup
 	intermediaryAccAddr := k.GetLockIdIntermediaryAccountConnection(ctx, lockID)
 	if intermediaryAccAddr.Empty() {
-		return nil, types.ErrNotSuperfluidUsedLockup
+		return types.ErrNotSuperfluidUsedLockup
 	}
 	intermediaryAcc := k.GetIntermediaryAccount(ctx, intermediaryAccAddr)
 	suffix := stakingSuffix(intermediaryAcc.ValAddr)
 
 	synthLock, err := k.lk.GetSyntheticLockup(ctx, lockID, suffix)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
 	if synthLock.Owner != sender {
-		return nil, lockuptypes.ErrNotLockOwner
+		return lockuptypes.ErrNotLockOwner
 	}
 
 	err = k.lk.DeleteSyntheticLockup(ctx, lockID, suffix)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
 	// use synthetic lockup coins for unbonding
 	amount := k.GetSuperfluidOSMOTokens(ctx, intermediaryAcc.Denom, synthLock.Coins.AmountOf(intermediaryAcc.Denom+suffix))
 
 	valAddr, err := sdk.ValAddressFromBech32(intermediaryAcc.ValAddr)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
 	shares, err := k.sk.ValidateUnbondAmount(
 		ctx, intermediaryAcc.GetAccAddress(), valAddr, amount,
 	)
+	if err != nil {
+		return err
+	}
 
+	res, err := k.sk.InstantUndelegate(ctx, intermediaryAcc.GetAccAddress(), valAddr, shares)
```

Comment:
;ets rename this `res` name to "undelegatedCoins"

Reply:
:+1: